include/mapbox/geojsonvt.hpp

```diff
@@ -60,6 +60,48 @@ inline uint64_t toID(uint8_t z, uint32_t x, uint32_t y) {
     return (((1ull << z) * y + x) * 32) + z;
 }
 
+class GeoJSONVTTile {
+public:
+    const Options options;
+
+    GeoJSONVTTile(const geojson& geojson_,
+                  uint8_t z,
+                  uint32_t x,
+                  uint32_t y,
+                  const Options& options_ = Options())
+        : GeoJSONVTTile(geojson::visit(geojson_, ToFeatureCollection{}), z, x, y, options_) {
+    }
+
+    GeoJSONVTTile(const mapbox::geometry::feature_collection<double>& features_,
+                  uint8_t z,
+                  uint32_t x,
+                  uint32_t y,
+                  const Options& options_)
+        : options(options_) {
+
+        auto z2 = std::pow(2, z);
+        auto tolerance = (options.tolerance / options.extent) / z2;
+        auto converted = detail::convert(features_, tolerance);
+        auto features = detail::wrap(converted, double(options.buffer) / options.extent);
+
+        const double z0 = 1u << z;
+        const double p =  options.buffer / options.extent;
+
+        const auto left = detail::clip<0>(features, (x - p) / z0, (x + 1 + p) / z0, -1, 2);
+        const auto clippedFeatures = detail::clip<1>(left, (y - p) / z0, (y + 1 + p) / z0, -1, 2);
+
+        internalTile.emplace(
+            detail::InternalTile{ clippedFeatures, z, x, y, options.extent, options.buffer, tolerance });
+    }
+
+    const Tile& getTile() {
+        return internalTile->tile;
+    }
+
+private:
+    std::experimental::optional<detail::InternalTile> internalTile;
```

Comment:
One more thing, does it have to be `optional`? I see that it's populated in constructor and then never removed dynamically, so perhaps we could do without `optional`?

Reply:
Doesn't need to be `optional`. In fixing this, I realized that this class doesn't need to maintain state and is better used as a method that returns a `Tile`